src/porepy/models/contact_mechanics_biot_model.py

```diff
@@ -0,0 +1,462 @@
+"""
+This is a setup class for solving the biot equations with contact between the fractures.
+
+The domain $[0, 2]\times[0, 1]$ with six fractures. We do not consider any fluid, and
+solve only for the linear elasticity coupled to the contact
+
+NOTE: This module should be considered an experimental feature, which will likely
+undergo major changes (or be deleted).
+
+"""
+import numpy as np
+import porepy as pp
+
+import porepy.models.contact_mechanics_model as contact_model
+from porepy.utils.derived_discretizations import implicit_euler as IE_discretizations
+
+
+class ContactMechanicsBiot(contact_model.ContactMechanics):
+    def __init__(self, mesh_args, folder_name):
+        super().__init__(mesh_args, folder_name)
+
+        # Temperature
+        self.scalar_variable = "p"
+        self.mortar_scalar_variable = "mortar_" + self.scalar_variable
+        self.scalar_coupling_term = "robin_" + self.scalar_variable
+        self.scalar_parameter_key = "flow"
+
+        # Scaling coefficients
+        self.scalar_scale = 1
+        self.length_scale = 1
+
+        # Time
+        # The time attribute may be used e.g. to update BCs.
+        self.time = 0
+        self.time_step = 1 * self.length_scale ** 2
```

Comment:
I don't think it should be moved to the ContactMechanics class (where there is no scalar variable). To be honest, I'm not sure what to do with it. On one hand, it is in many cases critical to avoid horrible condition number. On the other hand, it seems to be so closely linked to the particular problem that is solved, that it is tempting to remove it and leave it to the user.

Reply:
I think that is much better